### DISCUSSION TITLE
Redash memory consumption is too high

### DISCUSSION THREAD

**Discussion by aryankumar04** [getredash/redash#5971] [Q&A]
Deployed: AWS EC2
How: Dcoker-Compose
Ram: 64GB

Its frequently happening that memory usage suddenly steeps high upto max 64GB and EC2 Instance goes unreachable. 

How can we prevent this issue?

**Comment by justinclift** [comment]
Ahhh, that's a bit too generic of a problem report for really figuring out what's wrong.

How often are you experiencing the problem?  eg "once every few weeks" vs "once every few hours" (etc)

*Generally* the thing that causes high memory use is starting up too many worker processes.  With current versions of redash, each worker process seems to take about 1/4 of a GB of unshared RAM.

So, if you're starting up too many then it can go badly.

From your description though, it sounds more like memory use is ok for a bit... then is going suddenly very large.  That sounds more like a memory leak, or someone's running some kind of uncontained process.  Not sure what though. :frowning:
